pandas/core/reshape/reshape.py

```diff
@@ -586,19 +586,13 @@ def _stack_multi_columns(frame, level_num=-1, dropna=True):
     def _convert_level_number(level_num, columns):
         """
         Logic for converting the level number to something we can safely pass
-        to swaplevel:
-
-        We generally want to convert the level number into a level name, except
-        when columns do not have names, in which case we must leave as a level
-        number
+        to swaplevel. If `level_num` matches a column name return the name from
+        position `level_num`, otherwise return `level_num`.
```

Comment:
In general the first line of a docstring should be a one -liner. see https://numpydoc.readthedocs.io/en/latest/format.html#sections

(for internal functions, imo we could follow less strict guidelines, see #32773)